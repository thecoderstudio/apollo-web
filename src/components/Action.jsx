import React from 'react';
import styled from 'styled-components';

const Container = styled.div`
  display: grid;
  grid-template-columns: [image] 50px [content] 1fr;
  grid-template-rows: [title] 1fr [body] 1fr;
  grid-column-gap: 40px;
  align-items: center;
`;

const Image = styled.i`
  display: grid;
  justify-content: center;
  align-items: center;
  grid-column: image;
  grid-row: 1 / 3;
  height: 50px;
  width: 100%;
  color: ${props => props.theme.primary};
  border: 1px solid ${props => props.theme.primary};
  border-radius: 5px;
`;

const Title = styled.h4`
  grid-column: content;
  grid-row: title;
  color: ${props => props.theme.primary};
  margin: 0 0 8px 0;
`;

const Body = styled.p`
  grid-column: content;
  grid-row: body;
  margin: 0;
`;

function Action(props) {
  return (
    <Container>
      <Image className="material-icons">assessment</Image>
      <Title>{props.title}</Title>
      <Body>{props.children}</Body>
    </Container>
  );
}

export default Action;
