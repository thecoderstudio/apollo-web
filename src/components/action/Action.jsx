import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired
};

const Container = styled.div`
  display: grid;
  grid-template-columns: [image] 50px [content] 1fr;
  grid-template-rows: [title] 1fr [body] 1fr;
  grid-column-gap: 30px;
  align-items: center;
  margin: 16px;
`;

const Image = styled.i`
  display: grid;
  justify-content: center;
  align-items: center;
  grid-column: image;
  grid-row: 1 / 3;
  height: 50px;
  width: 100%;
  color: ${props => props.theme.primary};
  border: 1px solid ${props => props.theme.primary};
  border-radius: 5px;
`;

const Title = styled.h4`
  grid-column: content;
  grid-row: title;
  color: ${props => props.theme.primary};
  margin: 0 0 8px 0;
`;

const Body = styled.p`
  grid-column: content;
  grid-row: body;
  margin: 0;
`;

function Action(props) {
  const {title, children, ...rest} = props;
  return (
    <Container {...rest}>
      <Image className="material-icons">assessment</Image>
      <Title>{title}</Title>
      <Body>{children}</Body>
    </Container>
  );
}

Action.propTypes = propTypes;

export default Action;
