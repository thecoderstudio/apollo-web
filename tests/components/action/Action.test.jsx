import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Action from '../../../src/components/action/Action';

describe("Action", () => {
  it("renders correctly", () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Action title="test title" className="verify">test</Action>
      </BrowserRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
