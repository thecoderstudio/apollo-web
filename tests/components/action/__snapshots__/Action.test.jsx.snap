// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action renders correctly 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover {
  opacity: 0.90;
}

.c1 {
  display: grid;
  grid-template-columns: [image] 50px [content] 1fr;
  grid-template-rows: [title] 1fr [body] 1fr;
  grid-column-gap: 30px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 16px;
}

.c2 {
  display: grid;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-column: image;
  grid-row: 1 / 3;
  height: 50px;
  width: 100%;
  border: 1px solid;
  border-radius: 5px;
}

.c3 {
  grid-column: content;
  grid-row: title;
  margin: 0 0 8px 0;
}

.c4 {
  grid-column: content;
  grid-row: body;
  margin: 0;
}

<a
  className="c0 c1 verify"
  href="/agent/undefined/action/undefined"
  onClick={[Function]}
>
  <i
    className="c2 material-icons"
  >
    assessment
  </i>
  <h4
    className="c3"
  >
    test title
  </h4>
  <p
    className="c4"
  >
    test
  </p>
</a>
`;
